Xamarin.Essentials/Culture/Culture.android.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Globalization;
+using System.Threading;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Culture
+    {
+        static string PlatformInstalledUICulture =>
+            Java.Util.Locale.Default.ToString();
+
+        static void PlatformSetCurrentUICulture(CultureInfo cultureInfo)
+        {
+            Thread.CurrentThread.CurrentCulture = cultureInfo;
+            Thread.CurrentThread.CurrentUICulture = cultureInfo;
+        }
+
+        static CultureInfo PlatformGetCurrentUICulture(Func<string, CultureInfo> mappingOverride)
+        {
+            var netLanguage = ToDotnetLanguage(InstalledUICulture.Replace("_", "-"));
+
+            // this gets called a lot - try/catch can be expensive so consider caching or something
+            CultureInfo ci = null;
+            try
+            {
+                ci = new CultureInfo(netLanguage);
+            }
+            catch (CultureNotFoundException)
+            {
+                if (mappingOverride != null)
+                {
+                    return mappingOverride(InstalledUICulture);
+                }
+
+                // locale not valid .NET culture (eg. "en-ES" : English in Spain)
+                // fallback to first characters, in this case "en"
+                try
+                {
+                    var fallback = ToDotnetFallbackLanguage(new InternalCulture(netLanguage));
+                    ci = new CultureInfo(fallback);
+                }
+                catch (CultureNotFoundException)
+                {
+                    // language not valid .NET culture, falling back to English
+                    ci = new CultureInfo("en");
+                }
+            }
+            return ci;
+        }
+
+        static string ToDotnetLanguage(string androidLanguage)
+        {
+            var netLanguage = androidLanguage;
+
+            // certain languages need to be converted to CultureInfo equivalent
+            switch (androidLanguage)
```

Comment:
To support Hebrew, please add a case for "iw-IL" that will map to "he-IL"

Reply:
Yes, on Android Hebrew locale is called "iw-IL"

Reply:
added